main/src/library/MutList.flix

```diff
@@ -18,145 +18,407 @@
 /// Represents a mutable list.
 ///
 pub enum MutList[a] {
-    case MutList(Array[a], Int32)
+    case MutList(Ref[Array[a]], Ref[Int32])
 }
 
 namespace MutList {
 
-    /*
-     * Creation operations.
-     */
-
     ///
     /// Returns a new empty mutable list with a default capacity.
     ///
-    pub def new(): MutList[a] = newWithCapacity(0)
+    pub def new(): MutList[a] & Impure = MutList(ref [], ref 0)
 
     ///
-    /// Returns a new empty mutable list with the given `capacity`.
+    /// Returns a mutable list of all integers between `b` (inclusive) and `e` (exclusive).
+    ///
+    /// Returns an empty mutable list if `b >= e`.
     ///
-    pub def newWithCapacity(capacity: Int32): MutList[a] = ???
+    pub def range(b: Int32, e: Int32): MutList[Int32] & Impure = MutList(ref Array.range(b, e), ref (e - b))
 
-    /*
-     * Immutable operations.
-     */
+    ///
+    /// Returns the element at position `i` in the mutable list `v`.
+    ///
+    pub def get(i: Int32, v: MutList[a]): a & Impure =
+        let MutList(a, _) = v;
+        if (i > length(v))
+            get(-1, v)
```

Comment:
OK, this code is quiet clever. Took me a bit to understand it. I would suggest one change:

Replace `get(-1, v)` by `a[-1] // deliberately trigger index-out-of-bounds`